### PR TITLE
Update AWS S3 how-to with Dynamodb

### PR DESCRIPTION
**Pull request by Reshrahim**
Thank you for helping make the Radius documentation better!

**Please follow this checklist before submitting:**

- [ ] [Read the contribution guide](https://docs.radapp.io/community/contributing/docs/)
- [ ] Commands include options for Linux, MacOS, and Windows within codetabs
- [ ] New file and folder names are globally unique
- [ ] Page references use shortcodes instead of markdown or URL links
- [ ] Images use HTML style and have alternative text
- [ ] Places where multiple code/command options are given have codetabs

In addition, please fill out the following to help reviewers understand this pull request:

## Description
This PR updates the AWS S3 how-to with a Dynamodb as [s3 recently had breaking changes](https://github.com/radius-project/radius/issues/6535). Id has become the primary identifier and is read-only which makes the resource non-idempotent. We need to prioirtize and support [non-idmepotent resources](https://github.com/radius-project/radius/issues/6227) to unblock s3 ..until then a dynamo db example might be helpful.

Note that this is focused only on the modeling and deployment of AWS resource Dynamodb with hyperlinks to recipes and eshop tutorial which has coverage on app + aws resource

### Summary
🆕🚚🔄

This pull request replaces the how-to guide on using AWS resources with Radius to use AWS Dynamo DB instead of AWS S3. It adds new Bicep code and a sample application that demonstrate how to create and manage a Dynamo DB table with Radius. It also moves the old S3 guide and snippets to a holding-pen folder and updates the image names for the sample applications to use the new GitHub container registry.

> _Sing, O Muse, of the skillful pull request_
> _That changed the how-to guide of Radius_
> _From using S3, the cloud-born storage_
> _To Dynamo DB, the swift and scalable._

### Walkthrough
*  Refocus how-to guide on AWS Dynamo DB instead of AWS S3 ([link](https://github.com/radius-project/docs/pull/924/files?diff=unified&w=0#diff-e3c58ec3e1010335fbfe110c807c9220c058a2430e619feff2ba56a34909b4f5L13-R15), [link](https://github.com/radius-project/docs/pull/924/files?diff=unified&w=0#diff-e3c58ec3e1010335fbfe110c807c9220c058a2430e619feff2ba56a34909b4f5L55-R102), [link](https://github.com/radius-project/docs/pull/924/files?diff=unified&w=0#diff-e3c58ec3e1010335fbfe110c807c9220c058a2430e619feff2ba56a34909b4f5L121-R123))
*  Add Bicep code for modeling AWS Dynamo DB table ([link](https://github.com/radius-project/docs/pull/924/files?diff=unified&w=0#diff-aed284d404b959c74340140cf98ebef05b5e38189b94a264ff9b7a6a5c809900R1-R34))
*  Move original how-to guide for AWS S3 to holding-pen folder (`holding-pen/howto-aws-s3/index.md`, [link](https://github.com/radius-project/docs/pull/924/files?diff=unified&w=0#diff-199c2370218385096ffbf3d270af79acdc963dee744d3142e78ffbdf9d495bb2L1-R128), [link](https://github.com/radius-project/docs/pull/924/files?diff=unified&w=0#diff-3b9fab395dd2cd42fb2b2ca3dc38c2da1f2ca6b7e1c0bc5065de43a404881322))
*  Update image names for sample application containers to use GitHub container registry ([link](https://github.com/radius-project/docs/pull/924/files?diff=unified&w=0#diff-9cd4b3bfe8779a6f8c69ee7ee1defabc20ccee95ed1c859ba106c4b28640ce21L48-R48), [link](https://github.com/radius-project/docs/pull/924/files?diff=unified&w=0#diff-78b14eb57fca9bb204525401f01feaaa70865b12d5001b96302d27db896d8a46L49-R49))

### CODE REVIEW

File: docs/content/guides/author-apps/aws/howto-aws-resources/snippets/dynamodb.bicep [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        KeyType: 'HASH'
+      }
+      {
+        AttributeName: 'quantity'
+        KeyType: 'RANGE'
+      }
+    ]
+    Replicas: [
+     {
+      Region: 'us-west-2'
```
Comment (AaronCrawfis) [inline docs/content/guides/author-apps/aws/howto-aws-resources/snippets/dynamodb.bicep:29]: Could we make this a parameter?

File: docs/content/guides/author-apps/aws/howto-aws-resources/snippets/dynamodb.bicep [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+     } 
+    ]
+  }
+  alias: 'dynamodb'
```
Comment (AaronCrawfis) [inline docs/content/guides/author-apps/aws/howto-aws-resources/snippets/dynamodb.bicep:33]: Could we put alias at the top before properties as it serves a similar purpose to name in Azure/Radius resources?

File: docs/content/guides/author-apps/aws/howto-aws-resources/snippets/dynamodb.bicep [hunk 3]
```diff
@@ -0,0 +1,34 @@
+import aws as aws
+
```
Comment (AaronCrawfis) [inline docs/content/guides/author-apps/aws/howto-aws-resources/snippets/dynamodb.bicep:2]: Would there be an opportunity in this guide on showing how a user could access this dynamodb table from a container? Would that require an IAM user and roles and be complicated to setup? If so we may want to start prioritizing connections to AWS resources so we can complete the scenario of this guide. Without information on how to access this resource users will finish this up and not know fully how to proceed.

**Comment by Reshrahim** [comment]
The S3 issue is resolved from the AWS CloudFormation side. Closing this draft

**Pull request closed without merging**